core/src/main/java/com/linecorp/armeria/common/stream/PathStreamMessage.java

```diff
@@ -208,26 +225,30 @@ private final class PathSubscription implements Subscription, CompletionHandler<
 
         private final AsynchronousFileChannel fileChannel;
         private Subscriber<? super HttpData> downstream;
-        private final int bufferSize;
         private final EventExecutor executor;
+        private final int bufferSize;
+        private final long end;
         private final boolean notifyCancellation;
         private final boolean withPooledObjects;
 
         private boolean reading;
         private boolean closed;
 
         private volatile long requested;
-        private volatile int position;
+        private volatile long position;
 
         private PathSubscription(AsynchronousFileChannel fileChannel, Subscriber<? super HttpData> downstream,
-                                 EventExecutor executor, int bufferSize, boolean notifyCancellation,
-                                 boolean withPooledObjects) {
+                                 EventExecutor executor, long offset, long length, int bufferSize,
+                                 boolean notifyCancellation, boolean withPooledObjects) {
             this.fileChannel = fileChannel;
             this.downstream = downstream;
             this.executor = executor;
             this.bufferSize = bufferSize;
+            end = offset + length;
```

Comment:
https://github.com/line/armeria/blob/bc48105148d652e2e48e788c94bec007882c31dc/core/src/main/java/com/linecorp/armeria/common/stream/DefaultByteStreamMessage.java#L140

(Just curiotiy) @ikhoon , should we handle `overflow` here smilari with `DefaultByteStreamMessage`?
e.g offset(5) + lenth(Long.MAX_VALUE) = end(-xxx.., overflow) 

```java
assert position <= end; // if offset(5) + lenth(Long.MAX(), end is -xxx and this assertion failed
if (position < end) { ... }
```
on `L343`, we check `assert position <= end` so this assertion failed with overflow.

Reply:
Oops... it is a bug. 😱 Could you send a PR for this?

Reply:
oh~ I got it! I'll make fix PR with test ASAP 🙇

Reply:
Fix PR: https://github.com/line/armeria/pull/4705